src/models/component-ingredient.class.ts

```diff
@@ -3,23 +3,18 @@ import { ViviComponentFactory } from 'factory';
 import { Component } from './component.class';
 
 export class ComponentIngredient {
-    params: ComponentParams;
     component: Component;
 
     constructor(
-        private element: HTMLElement,
-        private factory: ViviComponentFactory<Component>
+        private parentEl: HTMLElement,
+        private factory: ViviComponentFactory<Component>,
+        private data?: ComponentParams
     ) {
-        // Get data from element
-        this.params = element.dataset;
+        this.component = this.factory.create(this.data);
+        this.component.append(this.parentEl, true);
     }
 
-    create() {
-        this.component = this.factory.create(this.params);
-        this.component.append(this.element.parentElement, true);
-    }
-
-    load(parent: Node) {
-        this.component.loadAll(parent);
+    load() {
+        this.component.loadAll();
     }
 }
```

Comment:
I wonder if the ingredients are even need at all any more? That would simplify some code if it could just straight go to creating components.

Reply:
Actually yeah. This cleans up a ton of unneeded complexity.
Done.